src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.tsx

```diff
@@ -79,32 +79,39 @@ export const Report = ({ activity, identifiers = [], versions = [] }: ReportProp
     ],
   });
 
-  const { execute: getAnswers } = useAsync(getAnswersApi);
-  const { execute: getLatestReport, isLoading: latestReportLoading } = useAsync(getLatestReportApi);
+  const { execute: getLatestReport, isLoading: latestReportLoading } = useAsync(
+    getLatestReportApi,
+    (response) => {
+      try {
+        const data = response?.data;
+        const headers = response?.headers;
+
+        if (data) {
+          const contentDisposition = headers?.['content-disposition'];
+          const fileName =
+            (contentDisposition &&
+              LATEST_REPORT_REGEX.exec(contentDisposition)?.groups?.filename) ??
+            LATEST_REPORT_DEFAULT_NAME;
+          const base64Str = Buffer.from(data).toString('base64');
+          const linkSource = getLatestReportUrl(base64Str);
+
+          download(linkSource, fileName, LATEST_REPORT_TYPE);
+        }
+      } catch (error) {
+        setLatestReportError(getErrorMessage(error));
```

Comment:
Do we need try catch here? Maybe just need to move setLatestReportError(getErrorMessage(error)) to the errorCallback of useAsync hook?

Reply:
no we need here